### ISSUE TITLE
Support new Authenticator system

### ISSUE THREAD

**Issue by darthf1** [auth0/jwt-auth-bundle#115]
In SF 5.1, a new Authenticator-based security system was [introduced](https://symfony.com/blog/new-in-symfony-5-1-updated-security-system) as an experimental feature. With SF 5.3 released (today), the Guard system is deprecated whilst the new Authenticator system is marked stable and will be set as the default for SF 6.0.

Upgrading to SF 5.3 results leads to the following deprecation notice:
```
The "Auth0\JWTAuthBundle\Security\Guard\JwtGuardAuthenticator" class extends "Symfony\Component\Security\Guard\AbstractGuardAuthenticator" that is deprecated since Symfony 5.3, use the new authenticator system instead.
```

Would be great if the new authenticator system can be supported.

More reading material here:
https://symfony.com/blog/new-in-symfony-5-3-guard-component-deprecation
https://symfony.com/doc/current/security/experimental_authenticators.html
https://symfony.com/doc/current/security/experimental_authenticators.html#creating-a-custom-authenticator

**Comment by evansims** [comment]
Thanks for the heads up on this, I'll investigate this when my bandwidth next opens up a little

**Comment by evansims** [comment]
📆  Scheduling this work for a new 5.0 milestone built on PHP SDK 8.0, pending its own beta release next month.

**Comment by abdojulari** [comment]
I have been trying to follow your article but I have a lot of issue. I am new to symfony, and integrating with auth0 is cumbersome. Jwt-auth-bundle has a lot of deprecation and I don't know how to fix it. I am running symfony 5.3

**Comment by emancypage** [comment]
We are looking for your MR @mcsky so lets ping @evansims to review your code.

**Comment by evansims** [comment]
Hi @emancypage 👋 I appreciate that you're waiting on this enhancement. As mentioned previously, other work currently has priority, but I'll review this PR as soon as I have bandwidth and am able to give it the attention it deserves. I appreciate your work on this @mcsky.

**Comment by evansims** [comment]
The changes provided by @mcsky have been merged into the new [5.x-dev branch](https://github.com/auth0/jwt-auth-bundle/tree/5.x-dev). Note that this branch should be considered experimental, but further feedback and pull requests are welcome.

As discussions about features to be included in the next major have become splintered across a few seperate issues, let's consolidate further discussion in this thread: https://github.com/auth0/jwt-auth-bundle/issues/130

Thanks very much for your contributions!
